example/src/main/java/org/secondbase/example/main/HelloAll.java

```diff
@@ -0,0 +1,58 @@
+package org.secondbase.example.main;
+
+import io.prometheus.client.Counter;
+import java.io.IOException;
+import org.secondbase.consul.ConsulModule;
+import org.secondbase.core.SecondBase;
+import org.secondbase.core.SecondBaseException;
+import org.secondbase.core.config.SecondBaseModule;
+import org.secondbase.flags.Flag;
+import org.secondbase.flags.Flags;
+import org.secondbase.logging.JsonLoggerModule;
+import org.secondbase.webconsole.HttpWebConsole;
+import org.secondbase.webconsole.PrometheusWebConsole;
+import org.secondbase.webconsole.widget.Widget;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+public class HelloAll {
+    @Flag(name="variable")
+    private static String var = "default";
+    @Flag(name="counter")
+    private static int counter = 1;
+
+    private static final Logger LOG = LoggerFactory.getLogger(HelloAll.class.getName());
+
+    public static void main(final String[] args) throws SecondBaseException, IOException {
+
+        final String[] realArgs = {
+                "--consul-host=localhost:8500",
+                "--service=HelloAll",
+                "--datacenter=local",
+                "--environment=testing",
+                "--webconsole-port=8000"
+        };
+
+        final Counter mycounter = Counter.build("mycounter", "a counter").register();
+
+        final Widget prometheusWidget = new PrometheusWebConsole();
+        final Widget[] widgets = {prometheusWidget};
+
+        final ConsulModule consul = new ConsulModule();
+        final SecondBaseModule webconsole = new HttpWebConsole(widgets);
+        final SecondBaseModule jsonLogger = new JsonLoggerModule();
+        final SecondBaseModule[] modules = {consul, webconsole, jsonLogger};
+
+        final Flags flags = new Flags().loadOpts(HelloAll.class);
+
+        new SecondBase(realArgs, modules, flags);
+
+        // Register custom service in consul
+        // TODO: This should be done automatically by the consul module. Moving common flags to
+        // SecondBase and using those flags from the init methods would allow this.
+        consul.registerServiceInConsul("HelloAll", 8000, "testing", "/healthz", 5L, "testtag");
```

Comment:
Maybe those should be flags in the consul module and the actual registration should work in the init module there?